hibernate-core/src/main/java/org/hibernate/sql/results/graph/collection/internal/DelayedCollectionInitializer.java

```diff
@@ -76,7 +76,9 @@ public void resolveKey(RowProcessingState rowProcessingState) {
 //						LoggingHelper.toLoggableString( getNavigablePath(), this.collectionKey.getKey() )
 //				);
 //			}
-			parentAccess.registerResolutionListener( owner -> collectionInstance.setOwner( owner ) );
+			if ( collectionInstance != null ) {
```

Comment:
Shouldn't this if move into the lambda instead? The field is only initialized when `resolveInstance` is called.

Reply:
I don't think so because the lamda firing is what yielded the NPE in the first place because `collectionInstance` was not yet resolved.  We can certainly rework it that way but you'd be adding a lamda that's firing for absolutely no reason.

Reply:
I went ahead and moved it into the lambda for now as requested.